### PR TITLE
Create task server connection

### PR DESCRIPTION
**Pull request by grammeaway**

### CODE REVIEW

File: src/api/task.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  },
+
+  /**
+   * Gets specifik task based on task id
+   * @param {Number} id
+   */
+  async get(id) {
+    const res = await request.get(`tasks/${id}`)
+    return await res.json()
+  },
```
Comment (kvartborg) [inline src/api/task.js]: Method overloading does not work in javascript a solution could be:
```js
async get(id) {
  const res = await request.get(id ? `tasks/${id}` : 'tasks')
  return await res.json()
}
```

File: src/api/task.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  /**
+   * Create new Task from Json body
+   * @param {Object} body
+   */
+  async create(body) {
+    const res = await request.post('tasks', body)
```
Comment (kvartborg) [inline src/api/task.js]: When #33 is merge, then adopt the superstruct for body validation.

File: src/api/task.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    return await res.json()
+  },
+
+  /**
+   * Update existing Task, based on Json body
+   * @param {Object} body
+   */
+  async update(body) {
+    const res = await request.update('tasks', body.json)
```
Comment (kvartborg) [inline src/api/task.js]: When #33 is merge, then adopt the superstruct for body validation.

Comment (kvartborg) [reply src/api/task.js]: Remove json part of the `body.json`.

File: src/views/Task/Create.js [hunk 4]
```diff
@@ -1,29 +1,96 @@
...
-    </label>
-    <input type="submit" value="Create Task" />
-  </Layout>
-)
+export default class Create extends Component {
+
+  state = {
+    startDate: moment(),
+    endDate: moment(),
+    title: null,
```
Comment (kvartborg) [inline src/views/Task/Create.js]: Maybe `""` is a better default value than `null`

File: src/views/Task/Create.js [hunk 5]
```diff
@@ -1,29 +1,96 @@
...
+
+  validateInput() {
+    //TODO
+
+  }
+
+  /**
+   * Sets the selected  date
+   */
+  updateStartDate = (date) => {
```
Comment (kvartborg) [inline src/views/Task/Create.js]: What about the date property, it should be documented in the docblock. 😄

File: src/views/Task/Create.js [hunk 6]
```diff
@@ -1,29 +1,96 @@
...
+          Description
+      <textarea onChange={this.handleChange} name="description"></textarea>
+        </label>
+        <label>
+          Urgency
+      <input type="text" onChange={this.handleChange} name="urgency" />
+        </label>
+        <input type="submit" value="Create Task" onClick={this.createTaskOnClick} />
+      </Layout>
+    )
```
Comment (kvartborg) [inline src/views/Task/Create.js]: There are some general indentation problems in the JSX, all inputs and datepickers should be indented so its aligned with the label text.

File: src/api/task.js (hunk 2 above)
Comment (grammeaway) [reply src/api/task.js]: Do we merge mine first, and then make a new PR when the superstruct works,, or should I just wait for the superstruckt?

File: src/views/Task/Create.js [hunk 7]
```diff
@@ -1,29 +1,99 @@
...
+   * Creates task based on data in inputfields
+   */
+  createTaskOnClick = () => {
+    const taskData = {
+      title: this.state.title,
+      description: this.state.description,
+      urgency: this.state.urgency,
+      startDate: this.state.startDate,
+      endDate: this.state.endDate
+    }
```
Comment (kvartborg) [inline src/views/Task/Create.js]: No need to create the object again, we could just pass the state directly to the `task.create` method.
```js
task.create(this.state)
```

File: src/views/Task/Create.js [hunk 8]
```diff
@@ -1,29 +1,99 @@
...
+        </label>
+        <label>
+          Description
+          <textarea onChange={this.handleChange} name="description"></textarea>
+        </label>
+        <label>
+          Urgency
+          <input type="text" onChange={this.handleChange} name="urgency" />
+        </label>
+          <input type="submit" value="Create Task" onClick={this.createTaskOnClick} />
```
Comment (kvartborg) [inline src/views/Task/Create.js]:
```suggestion
         <input type="submit" value="Create Task" onClick={this.createTaskOnClick} />
```

File: src/views/Task/Create.js [hunk 9]
```diff
@@ -1,29 +1,160 @@
...
+    try {
+      const taskData = {
+        title: this.state.title,
+        description: this.state.description,
+        urgency: this.state.urgency,
+        start: this.state.startDate.toDate(), //toDate() to convert moment()-date to standard JS-date, due to Superstruckt and server limitations
+        deadline: this.state.endDate.toDate() //toDate() to convert moment()-date to standard JS-date, due to Superstruckt and server limitations
+      }
+      await task.create(taskData)
+      //Redirect to employers task-overview
```
Comment (kvartborg) [inline src/views/Task/Create.js]: Take a look at #39, to see how redirect is implemented

File: src/views/Task/Create.js [hunk 10]
```diff
@@ -1,29 +1,160 @@
...
+        <section>
+          <h2>Create Task</h2>
+          {this.state.error && (
+            <Alert>{this.state.error} </Alert>
+          )}
+          <form onSubmit={this.submitHandler}>
+            {this.fields.map(this.fieldRender.bind(this))}
+            <label>
+            Starting Date
+            <DatePicker onChange={this.updateStartDate} minDate={moment()} selected={this.state.startDate} />
```
Comment (kvartborg) [inline src/views/Task/Create.js]: Indent one space

File: src/views/Task/Create.js [hunk 11]
```diff
@@ -1,29 +1,160 @@
...
+            </label>
+            <label>
+            Deadline
+            <DatePicker onChange={this.updateEndDate} minDate={this.state.startDate} selected={this.state.endDate} />
```
Comment (kvartborg) [inline src/views/Task/Create.js]: Indent one space

File: src/views/Task/Create.js [hunk 12]
```diff
@@ -1,29 +1,162 @@
...
-    <label>
-      Urgency
-      <input type="text" />
-    </label>
-    <input type="submit" value="Create Task" />
-  </Layout>
-)
+export default class Create extends Component {
+
+  //auth.login("username", "password")
```
Comment (NJohansen) [inline src/views/Task/Create.js]: If comment shouldnt be there remove it.

File: src/views/Task/Create.js [hunk 13]
```diff
@@ -1,29 +1,162 @@
...
+    startDate: moment(),
+    endDate: moment(),
+    title: "",
+    description: "",
+    urgency: "",
+    error: null
+  }
+
+  fields = [
+    //["label", "Type", Validation method]
```
Comment (NJohansen) [inline src/views/Task/Create.js]: If comment shouldnt be there remove it.

File: src/views/Task/Create.js [hunk 14]
```diff
@@ -1,29 +1,162 @@
...
+    ["Urgency", "text", v => v.length > 0, false]
+  ]
+
+
+  /**
+   * Creates task based on data in inputfields
+   */
+  submitHandler = async event => {
+    event.preventDefault()
+    if (!this.validator()) { //invalid input case
```
Comment (NJohansen) [inline src/views/Task/Create.js]: Where should this comment be?

File: src/views/Task/Create.js [hunk 15]
```diff
@@ -1,29 +1,162 @@
...
+      this.setState({ error: "Invalid input" })
```
Comment (kvartborg) [inline src/views/Task/Create.js]: Missing return

File: src/views/Task/Create.js [hunk 16]
```diff
@@ -1,29 +1,162 @@
...
+      await task.create(taskData)
+      //Redirect to employers task-overview
+    } catch (e) {
+      this.setState({ error: e.message })
+    }
+  }
+
+  /**
+   * Validates input based on validation mehotd defined in Fields[]
+   * @return {bolean}
```
Comment (NJohansen) [inline src/views/Task/Create.js]: Spelling mistake

**Pull request merged**